src/com/sun/jna/Memory.java

```diff
@@ -137,71 +65,17 @@ public static void purge() {
 
     /** Dispose of all allocated memory. */
     public static void disposeAll() {
-        synchronized (LinkedReference.class) {
-            LinkedReference entry;
-
-            while ((entry = HEAD) != null) {
-                Memory memory = HEAD.get();
-
-                if (memory != null) {
-                    // dispose does the unlink call internal
-                    memory.dispose();
-                } else {
-                    HEAD.unlink();
-                }
-
-                if (HEAD == entry) {
-                    throw new IllegalStateException("the HEAD did not change");
-                }
-            }
-        }
-
-        synchronized (QUEUE) {
-            LinkedReference stale;
-
-            // try to release as mutch memory as possible
-            while ((stale = (LinkedReference) QUEUE.poll()) != null) {
-                stale.unlink();
-            }
-        }
-    }
-
-    /**
-     * Unit-testing only, ensure the doubly linked list is in a good shape.
-     *
-     * @return the number of tracked instances
-     */
-    static int integrityCheck() {
-        synchronized (LinkedReference.class) {
-            if (HEAD == null) {
-                return 0;
-            }
-
-            ArrayList<LinkedReference> entries = new ArrayList<LinkedReference>();
-            LinkedReference entry = HEAD;
-
-            while (entry != null) {
-                entries.add(entry);
-                entry = entry.next;
-            }
-
-            int index = entries.size() - 1;
-            entry = entries.get(index);
-
-            while (entry != null) {
-                if (entries.get(index) != entry) {
-                    throw new IllegalStateException(entries.get(index) + " vs. " + entry + " at index " + index);
-                }
-
-                entry = entry.prev;
-                index--;
+        // use a copy since dispose() modifies the map
+        Collection<Reference<Memory>> refs = new LinkedList<Reference<Memory>>(allocatedMemory.values());
```

Comment:
Since this is just a temporary variable is there any value in using `LinkedList` vs. `ArrayList`?

Reply:
Valid point pushed updated version.